fix: made JWT typ header property optional in Operate and Tasklist

kristinkomschow opened the pull request:
## Description

<!-- Describe the goal and purpose of this PR. -->
Added `null` as possible type value for the JWT header to make the property optional (in accordance with [the JWT spec](https://datatracker.ietf.org/doc/html/rfc7519#section-5.1)).

I also applied a fix to the Tasklist decoder that we applied to Operate some weeks ago related to the decoder not being able to handle JWT with algorithms other than RS256 as I expect this would become an issue for the same customer in the near future (Operate issue about this: https://github.com/camunda/camunda/issues/23727)

Will be backported to 8.7, 8.6 and 8.5.

## Checklist

<!--- Please delete options that are not relevant. Boxes should be checked by reviewer. -->
- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)
  - [ ] enable backports [when recommended](https://github.com/camunda/camunda/wiki/CI-&-Automation#when-to-backport-ci-changes)

## Related issues

relates to https://github.com/camunda/camunda/issues/29314

kristinkomschow commented:
The first commit fixes a buggy assertion, as noted here: https://github.com/camunda/camunda/pull/29506#issuecomment-2722361414

nathansandi commented on tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/security/oauth/IdentityOAuth2WebConfigurer.java:
```diff
@@ -70,10 +84,20 @@ public void configure(final HttpSecurity http) throws Exception {
     return NimbusJwtDecoder.withJwkSetUri(getJwkSetUriProperty())
+        .jwsAlgorithms(
+            algorithms -> {
+              algorithms.add(RS256);
```
```suggestion
+List.of(RS256, RS384, RS512, ES256, ES384, ES512).forEach(algorithms::add);
```

kristinkomschow commented:
I added all of the suggestions and squashed them into existing commits where it fit (to have a cleaner history).
Also refactored the Operate tests to use parameterized tests as it is a lot cleaner. Thanks for that suggestion!

I'm gonna ask the team about the failing tests if they still happen now (I also rebased on main).

backport-action commented:
Successfully created backport PR for `stable/8.6`:
- #29830

backport-action commented:
Successfully created backport PR for `stable/8.7`:
- #29831
